From f65aca4112b69e13fec631bc1c753fe7f19af173 Mon Sep 17 00:00:00 2001
From: Umashankar Godachi <umasha@codeaurora.org>
Date: Mon, 6 Nov 2017 20:57:07 +0530
Subject: [PATCH 4/4] Fix non-protected broadcasts sent from phone process.

Intents broadcasted from system internals should be
protected inorder to avoid security holes. So fix is
to make the phone process intents broadcasted from
proprietary space to be protected.

Change-Id: Ie9cf2e5923b8a52e9b3ba91e9e9c3d649aae9fca
CRs-Fixed: 2128245
---
 AndroidManifest.xml | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 2602e465c..78b7e120f 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -90,6 +90,15 @@
     <protected-broadcast android:name= "android.telephony.action.SIM_APPLICATION_STATE_CHANGED" />
     <protected-broadcast android:name= "android.telephony.action.SIM_SLOT_STATUS_CHANGED" />
     <protected-broadcast android:name= "android.telephony.action.SUBSCRIPTION_CARRIER_IDENTITY_CHANGED" />
+    <protected-broadcast android:name= "codeaurora.intent.action.ACTION_MANAGED_ROAMING_IND" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.SUBSCRIPTION_INFO_RECORD_ADDED" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.PRIMARY_CARD_CHANGED_IN_SERVICE" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.ACTION_SET_PRIMARY_CARD_DONE" />
+    <protected-broadcast android:name= "codeaurora.intent.action.ACTION_LTE_CONFIGURE" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.ACTION_UICC_MANUAL_PROVISION_STATUS_CHANGED" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.ACTION_RADIO_CAPABILITY_UPDATED" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.ACTION_NETWORK_SPECIFIER_SET" />
+    <protected-broadcast android:name= "org.codeaurora.intent.action.ACTION_DDS_SWITCH_DONE" />
 
     <uses-permission android:name="android.permission.BROADCAST_STICKY" />
     <uses-permission android:name="android.permission.CALL_PHONE" />
-- 
2.17.1

